npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb

```diff
@@ -453,6 +433,41 @@ def lockfile_dependencies(lockfile)
             ).parse
         end
 
+        def raise_resolvability_error_if_necessary(error_message, yarn_lock)
+          sub_dep_local_path_err = "refers to a non-existing file"
+          if error_message.match?(INVALID_PACKAGE) ||
+             error_message.include?(sub_dep_local_path_err)
+            raise_resolvability_error(error_message, yarn_lock)
+          end
+        end
+
+        def raise_missing_package_error_if_necessary(error, yarn_lock)
+          error_message = error.message
+
+          if error_message.include?("Couldn't find package")
+            package_name = error_message.match(/package "(?<package_req>.*?)"/)
+                                        .named_captures["package_req"]
+                                        .split(/(?<=\w)\@/).first
+            sanitized_name = sanitize_package_name(package_name)
+            sanitized_error = error_message.gsub(package_name, sanitized_name)
+            handle_missing_package(sanitized_name, sanitized_error, yarn_lock)
+
+          elsif error_message.match?(%r{/[^/]+: Not found})
+            package_name = error_message
+                           .match(%r{/(?<package_name>[^/]+): Not found})
+                           .named_captures["package_name"]
+            sanitized_name = sanitize_package_name(package_name)
+            sanitized_error = error_message.gsub(package_name, sanitized_name)
+            handle_missing_package(sanitized_name, sanitized_error, yarn_lock)
+
+          elsif error_message.include?("Tarball is not in network and can not be located in cache")
```

Comment:
please can we add unit tests too?

Reply:
After completing the Yarn Error code implementation I am planing to implement spec for that.